### PR TITLE
Extend migrate to handle some more use cases 

### PR DESCRIPTION
**Pull request by Tatskaari**
Mostly refactoring the code so it can handle more edge cases. I've also added some tests that cover most of the scenarios I've encountered when trying to run this on our codebase. ~~It's still not quite there but is getting close.~~ It's there! 

This tool now:
- Generates aliases for binary targets correctly 
- Supports multiple third party directories, leaving aliases behind for the targets that used to live there
- Has a control flow that makes some modicum of sense

### CODE REVIEW

File: graph/graph.go [hunk 1]
```diff
@@ -138,7 +143,7 @@ func (g *Graph) ensureVisibility(conf *config.Config, dep *Dependency) error {
 
 	kind := conf.GetKind(t.Kind())
 	if kind == nil {
-		return fmt.Errorf("%v tried to depend on %v, but %v is of unknown kind: %v", dep.From.Format(), dep.To.Format(), dep.To.Format(), t.Kind())
+		return nil
```
Comment (Tatskaari) [inline graph/graph.go:146]: It's better to just assume the kind is visible to the target rather than giving up.

File: puku.go [hunk 2]
```diff
@@ -85,7 +85,7 @@ var funcs = map[string]func(conf *config.Config, plzConf *please.Config, orignal
 }
 
 func main() {
-	flags.ParseFlagsOrDie("puku", &opts, nil)
+	cmd := flags.ParseFlagsOrDie("puku", &opts, nil)
```
Comment (Tatskaari) [inline puku.go:88]: we were double parsing which meant the paths were doubled up. This broke the migrator.

File: .golangci.yml [hunk 3]
```diff
@@ -14,7 +14,6 @@ linters:
     - dupl
     - exportloopref
     - gci
-    - gocritic
```
Comment (Tatskaari) [inline .golangci.yml:17]: This guy was making some pretty debatable recommendations so I nuked it.

**Review by samwestmoreland — APPROVED** [review]
